src/IR/LowerAst.hs

```diff
@@ -162,6 +162,11 @@ lowerExpr (A.OpRegion _ _) =
 lowerExpr (A.Match s ps) =
   I.Match <$> lowerExpr s <*> mapM lowerArm ps <*> pure untyped
   where lowerArm (a, e) = (,) <$> lowerPatAlt a <*> lowerExpr e
+lowerExpr (A.Tuple es) =
+  apply_recurse (I.Data "Pair" untyped) <$> mapM lowerExpr es
```

Comment:
FYI: this is the wrong data constructor name. You should use the `tupleId` helper to derive the name of the tuple: https://github.com/ssm-lang/sslang/blob/main/src/IR/Types/Type.hs#L224

Reply:
If I also change my inserted data type to the name getting from tupleId, it will generate errors from scope "NameError (ErrorMsg "'(,)' should begin with upper case or begin and end with ':'")". I guess it is because naming it "(,)" might not pass the naming check.

Reply:
Ah ok, I should fix that then.